### PR TITLE
Reduces the copypaste in cleanable decal code

### LINKED ISSUE #20294: Metal rods display the wrong stack icon upon creation
When you make multiple metal rods, let's say 10 at a time, it shows the sprite for only one rod, despite the stack containing a lot more. Only after using the rods does the sprite refresh (via update_icon I assume) and show the sprite for multiple rods.

### PR DESCRIPTION
**Pull request by ChangelingRain**
Fixes #20294 kind of incidentally

### CODE REVIEW

**Comment by WJohn** [comment]
:+1: Thank god this was such bullshit to deal with. Drive-by-anti-cult borgs were annoying as all hell.

**Comment by Supermichael777** [comment]
powercreap
then again janiborgs only have clean on walkover because THEY DON'T HAVE A BUCKET for the mop

**Comment by KorPhaeron** [comment]
I'm not sure the only two counters for anything the cult does should be tied to a single civilian job that quietly goes missing most of the time anyway

**Comment by Alek2ander** [comment]
Chaplain should be able to bless borgs then. The machine gods must be appeased

**Comment by ChangelingRain** [comment]
Kor you know that's not totally true, the chaplain also exists.
But if your victory against the cult is really riding on a janitor(who probably doesn't have holy water) and a not-particularly-common borg type you're probably already fucked.

**Comment by Shapsy** [comment]
I think he was talking about the Chaplain and how this puts even more on his shoulders

**Comment by Supermichael777** [comment]
yeah your strawmaning

unless you add an investiture function to hop you need to make him strong but not the only one

and dont give me that you can order it from cargo crap thats never enough

**Comment by AdamElTablawy** [comment]
make chaplain able to slap janiborgs with bible to turn their cleaning into HOLY cleaning

**Comment by KorPhaeron** [comment]
For everyone upvoting, what is security supposed to do if the cult has a network of tele runes all over the place and the chaplain died/never signed up?

**Comment by PKPenguin321** [comment]
Build a girder over the runes, at least once joan's PR gets merged

**Comment by optimumtact** [comment]
Kor pretending that security doesn't just kill cultists and tries to clean runes instead is pretty trippy.

**Comment by KorPhaeron** [comment]
@optimumtact 

You gotta clean their runes to limit their movement so you can kill them

**Comment by optimumtact** [comment]
Kor said no so you should probably refactor this into something else

**Comment by optimumtact** [comment]
Or close it

File: code/game/gamemodes/miniantags/slaughter/slaughter.dm [hunk 1]
```diff
@@ -59,12 +58,6 @@
 	else
 		speed = 0
 
-/obj/effect/decal/cleanable/blood/innards
-	icon = 'icons/obj/surgery.dmi'
-	name = "pile of viscera"
-	desc = "A repulsive pile of guts and gore."
-	random_icon_states = list("innards")
```
Comment (phil235) [inline code/game/gamemodes/miniantags/slaughter/slaughter.dm]: undocumented changes

File: code/game/turfs/open.dm [hunk 2]
```diff
@@ -88,7 +88,7 @@
 
 	clean_blood()
 	for(var/obj/effect/O in src)
-		if(is_cleanable(O))
+		if(istype(O, /obj/effect/decal/cleanable))
 			qdel(O)
```
Comment (phil235) [inline code/game/turfs/open.dm]: this means obj/effect/rune are not removed by this anymore

File: code/modules/mob/living/silicon/robot/robot.dm [hunk 3]
```diff
@@ -785,9 +785,8 @@
 			if(isturf(tile))
 				tile.clean_blood()
 				for(var/A in tile)
-					if(istype(A, /obj/effect))
-						if(is_cleanable(A))
-							qdel(A)
+					if(istype(A, /obj/effect/decal/cleanable))
+						qdel(A)
```
Comment (phil235) [inline code/modules/mob/living/silicon/robot/robot.dm]: this means obj/effect/rune are not removed by this anymore

File: code/modules/vehicles/pimpin_ride.dm [hunk 4]
```diff
@@ -47,7 +47,7 @@
 		if(isturf(tile))
 			tile.clean_blood()
 			for(var/A in tile)
-				if(is_cleanable(A))
+				if(istype(A, /obj/effect/decal/cleanable))
```
Comment (phil235) [inline code/modules/vehicles/pimpin_ride.dm]: this means obj/effect/rune are not removed by this anymore.

File: code/modules/vehicles/pimpin_ride.dm [hunk 5]
```diff
@@ -47,7 +47,7 @@
 		if(isturf(tile))
 			tile.clean_blood()
 			for(var/A in tile)
-				if(is_cleanable(A))
+				if(iscleanable(A))
```
Comment (phil235) [inline code/modules/vehicles/pimpin_ride.dm]: typo?

File: code/modules/mob/living/silicon/robot/robot.dm [hunk 6]
```diff
@@ -785,9 +785,8 @@
 			if(isturf(tile))
 				tile.clean_blood()
 				for(var/A in tile)
-					if(istype(A, /obj/effect))
-						if(is_cleanable(A))
-							qdel(A)
+					if(iscleanable(A))
```
Comment (phil235) [inline code/modules/mob/living/silicon/robot/robot.dm]: typo?

File: code/game/turfs/open.dm [hunk 7]
```diff
@@ -88,7 +88,7 @@
 
 	clean_blood()
 	for(var/obj/effect/O in src)
-		if(is_cleanable(O))
+		if(iscleanable(O))
```
Comment (phil235) [inline code/game/turfs/open.dm]: typo?

**Review by phil235 — APPROVED** [review]
:+1:

File: code/game/objects/effects/decals/cleanable.dm [hunk 8]
```diff
@@ -1,4 +1,6 @@
 /obj/effect/decal/cleanable
+	gender = PLURAL
```
Comment (optimumtact) [inline code/game/objects/effects/decals/cleanable.dm]: Why set all the others to NEUTER and this one to PLURAL?

**Pull request merged**